File: java/src/main/java/exercises/streams/StandardDeviation.java
package exercises.streams;

import java.util.Collection;

public class StandardDeviation {
    public static double calculateStdDeviation(Collection<Long> values) {
        return Math.sqrt(getSumOfDiffsSquared(values, getAverage(values)));
    }

    private static double getSumOfDiffsSquared(Collection<Long> values, Double average) {
        double sum = 0.0;
        for (Long value : values) {
            sum += Math.pow((value - average), 2);
        }
        return sum / values.size();
    }

    private static double getAverage(Collection<Long> values) {
        double sum = 0.0;
        for (Long value : values) {
            sum += value;
        }
        return sum / values.size();
    }
}

File: assigment/c#/assignment-3/NameStats/StandardDeviation.cs
using System;
using System.Collections.Generic;

namespace NameStats
{
    public class StandardDeviation
    {
        public static double CalculateStandardDeviation(IList<long> values)
        {
            return Math.Sqrt(GetSumOfDiffsSquared(values, GetAverage(values)));
        }

        private static double GetSumOfDiffsSquared(IList<long> values, double average)
        {
            double sum = 0.0d;
            foreach (var value in values)
            {
                sum += Math.Pow((value - average), 2);
            }

            return sum / values.Count;
        }

        private static double GetAverage(IList<long> values)
        {
            double sum = 0.0d;
            foreach (var value in values)
            {
                sum += value;
            }

            return sum / values.Count;
        }
    }
}